### ISSUE TITLE
Android - Autofill credit card not filled after enter CVC

### ISSUE THREAD

**Issue by hungcaovn** [flutter/flutter#145738]
### Steps to reproduce

I tested with the example code from this link: https://api.flutter.dev/flutter/widgets/AutofillGroup-class.html
In the "Credit Card Information" field, after select the saved CC, enter CVC in the popup then nothing happen. There is not anything in the CC field
***Note: Tested with Google Pixel 3 emulator

### Expected results

CC details should be auto filled

### Actual results

Nothing is filled

### Code sample

```
AutofillGroup(
          child: Column(
            children: <Widget>[
              TextField(
                controller: creditCardNumber,
                keyboardType: TextInputType.number,
                autofillHints: const <String>[AutofillHints.creditCardNumber],
              ),
              TextField(
                controller: creditCardSecurityCode,
                autofillHints: const <String>[
                  AutofillHints.creditCardSecurityCode,
                ],
              ),
            ],
          ),
        ),
```

### Screenshots or Video

![Screenshot_1711430250](https://github.com/flutter/flutter/assets/13977238/13ac7926-2f53-4bca-8ab9-9fad36f996ab)

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

[✓] Flutter (Channel stable, 3.13.8, on macOS 14.1.1 23B81 darwin-arm64, locale en-VN)
    • Flutter version 3.13.8 on channel stable at /Users/hungcao/Documents/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 6c4930c4ac (5 months ago), 2023-10-18 10:57:55 -0500
    • Engine revision 767d8c75e8
    • Dart version 3.1.4
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/hungcao/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.14.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.87.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.84.0

[✓] Connected device (4 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554                        • android-arm64  • Android 12 (API
      32) (emulator)
    • iPhone 15 Pro (mobile)      • D4B9594A-3552-4D17-833E-63CEF316A059 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-17-0 (simulator)
    • macOS (desktop)             • macos                                • darwin-arm64   • macOS 14.1.1
      23B81 darwin-arm64
    • Chrome (web)                • chrome                               • web-javascript • Google Chrome
      123.0.6312.59

[✓] Network resources
    • All expected network resources are available.

• No issues found!

**Comment by darshankawar** [comment]
@hungcaovn 
I think you are hitting https://github.com/flutter/flutter/issues/116780 so please follow-up in it for further updates. You may share your findings in that issue for better tracking.
Closing this as a duplicate. If you disagree, write in comments and I'll reopen it.

**Comment by hungcaovn** [comment]
@darshankawar I don't think it is the same as https://github.com/flutter/flutter/issues/116780. It worked fine with another Android phone (Nokia) but did not work on Google Pixel 3. The credit card suggestion shows up but it is not filled in the text field after enter enter CVC

**Comment by darshankawar** [comment]
Thanks for feedback. Reopening.

**Comment by darshankawar** [comment]
@hungcaovn 
Although I don't have Pixel device to verify this, but I tried on Samsung S10 and seem to be hitting https://github.com/flutter/flutter/issues/116780 wherein, the credit card info is not pulled properly.
Is there a way for you to check also on another device (you mentioned Nokia, but also any other one) on which it works or not ?
Also, please upgrade to latest stable and re-run your scenario.

**Comment by hungcaovn** [comment]
@darshankawar : Okay I will try to upgrade flutter. I attached the video for better clarifying 

https://github.com/flutter/flutter/assets/13977238/b1f20c72-db6f-481e-8f23-5acc6cd117f9

**Comment by hungcaovn** [comment]
@darshankawar : Tried with the latest Flutter version (v3.19.4) but got the same issue. I tried with Nexus emulator (I don't have many Android devices to test). Also, if I enter something in the text field and clear it, then select suggested CC number and it works as expected. I believed there was something wrong with CVC stuff. Check below video:
https://github.com/flutter/flutter/assets/13977238/68012906-35ce-426e-b633-df60359f333f

**Comment by hungcaovn** [comment]
For anyone getting this issue like me, I found a workaround: Set launchMode="standard" in AndroidManifest

**Comment by letsar** [comment]
I have the same issue, and it seems that it's related to the fact that the connection with the `TextInput` is closed while the system popup is shown. Is it possible to reopen this issue and investigate further? (Setting launchMode="standard" not working).

**Comment by hungcaovn** [comment]
@letsar : Yes, I found that after entered the CVC, app is resumed and somehow it could not continue with the autofill process

**Comment by hungcaovn** [comment]
Reopen this issue because setting launchMode="standard" did not resolve it permanently

**Comment by darshankawar** [comment]
Thanks for the update. I was unable to replicate this on S10 with Android 12 version using latest version, but based on the report and behavior, I'll keep the issue open for team's input / attention.

**Comment by zoeyfan** [comment]
Cc myself to follow this... @zoeyfan

**Comment by justinmc** [comment]
I wonder if this is as simple as the field losing focus when the CVC popup shows up. In the video, you can just see in the background that the credit card field seems to lose focus at the time that the CVC popup appears. Confirmed with @LongCatIsLooong that autofill depends on a text field being focused for it to work.

**Comment by hungcaovn** [comment]
> I wonder if this is as simple as the field losing focus when the CVC popup shows up. In the video, you can just see in the background that the credit card field seems to lose focus at the time that the CVC popup appears. Confirmed with @LongCatIsLooong that autofill depends on a text field being focused for it to work.

@justinmc : Yes maybe that is the cause. But I am not sure why it works on some devices and not on the others. Even on the same device sometimes it works and sometimes not

**Comment by justinmc** [comment]
I wasn't able to reproduce this on my physical Pixel 6, but maybe that's because it uses a fingerprint verification instead of a CVC field. The credit card number field did not appear to lose focus during that time.

<details>

<summary>Full repro code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            AutofillGroup(
              child: Column(
                children: <Widget>[
                  TextField(
                    keyboardType: TextInputType.number,
                    autofillHints: const <String>[
                      AutofillHints.creditCardNumber,
                    ],
                    inputFormatters: [
                      FilteringTextInputFormatter.digitsOnly,
                      LengthLimitingTextInputFormatter(19),
                    ],
                    decoration: const InputDecoration(
                      hintText: 'Credit card number',
                    ),
                  ),
                  TextField(
                    decoration: const InputDecoration(
                      hintText: 'Credit card security code',
                    ),
                    obscureText: true,
                    inputFormatters: [
                      FilteringTextInputFormatter.digitsOnly,
                      LengthLimitingTextInputFormatter(4),
                    ],
                    autofillHints: const <String>[
                      AutofillHints.creditCardSecurityCode,
                    ],
                  ),
                ],
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

**Comment by minhnguyenandpad** [comment]
I got this issue too in the latest flutter version.

**Comment by justinmc** [comment]
In support of the focus theory, a similar problem seems to happen with autofill on Windows: https://github.com/flutter/flutter/issues/162066

Also I tried a few native experiences and I see that it varies on whether or not the field remains focused:

| Platform | When autofilling a credit card number with fingerprint... |
| --- | --- |
| Android Amazon App | The field remains focused during fingerprint |
| Netflix website on Android Chrome | The field loses focus during fingerprint |

**Comment by camsim99** [comment]
Just tried this on a Galaxy S21 5G running Android 14 with autofill from Google Pay and it seemed to work for me, so can confirm this is device-specific.

**Comment by camsim99** [comment]
Did another test on Pixel 3A running Android 12 and I also see it working. I tried a couple of different times, but I know reporters said it's inconsistent so I should probably test this and the Galaxy a bit more. Next up, I'll try some older devices and emulators.

**Comment by justinmc** [comment]
I also couldn't repro on a physical Pixel 9, regardless of whether it asked me for the CVC and fingerprint or just the fingerprint.

**Comment by camsim99** [comment]
Just tested a Pixel 7 API 32 emulator and upon click of the text field, it actually showed a dropdown menu instead of hints on top of the IME. The dropdown did work. I've noticed we've mostly been testing Pixels, so I'll definitely try another (older) Samsung when I get the chance.

**Comment by camsim99** [comment]
Just tested a Samsung Galaxy J7 running Android 8.1 (API 27) and upon click of the text field, it actually showed a dropdown menu instead of hints on top of the IME. The dropdown did work.

**Comment by camsim99** [comment]
To summarize all my testing (I'll just edit this table from here on out):

| Device  | Android Version | IME hint or Dialog above cursor| Did credit card number and CVC autofill?
| ------------- | ------------- | --- | ----|
| Samsung S21  | Android 14 (API 34)  | IME hint | ✅  |
| Pixel 3A | Android 12 (API 32) | IME hint | ✅  |
| Pixel 7 emulator | Android 12 (API 32)  | Dialog | ✅  |
| Samsung Galaxy J7 | Android 8.1 (API 27)  | Dialog | ✅  |

**Comment by camsim99** [comment]
Closing this issue as I have not been able to repro. If you're still seeing this issue, please leave details to help us repro if you can. Thanks!
